nifi-asymmetrik-standard-bundle/nifi-asymmetrik-standard-processors/src/test/java/com/asymmetrik/nifi/processors/aws/ListS3BatchTest.java

```diff
@@ -0,0 +1,302 @@
+package com.asymmetrik.nifi.processors.aws;
+
+import org.apache.nifi.components.state.Scope;
+import org.apache.nifi.controller.AbstractControllerService;
+import org.apache.nifi.processor.ProcessContext;
+import org.apache.nifi.processor.exception.ProcessException;
+import org.apache.nifi.processors.aws.credentials.provider.service.AWSCredentialsProviderService;
+import org.apache.nifi.util.MockFlowFile;
+import org.apache.nifi.util.TestRunner;
+import org.apache.nifi.util.TestRunners;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+import com.amazonaws.auth.AWSCredentialsProvider;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.model.ListObjectsV2Request;
+import com.amazonaws.services.s3.model.ListObjectsV2Result;
+import com.amazonaws.services.s3.model.S3ObjectSummary;
+import com.google.common.collect.ImmutableList;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+import static com.asymmetrik.nifi.processors.aws.ListS3Batch.*;
+
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class ListS3BatchTest {
```

Comment:
Don't think it should hold up the PR, but some test cases where no S3 results are returned would be nice

Reply:
Might also be useful to have test cases with non-ascii file names?

Reply:
Done!